### DISCUSSION TITLE
268

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#756] [Comments]
http://jojoldu.tistory.com/268

**Comment by jojoldu** [comment]
작성자: 카엔

좋은 툴 소개 정말 감사드립니다!
안그래도 Svn쓰다가 Git으로 넘어와서 애먹고 있는데 이 툴 보니까 정말 좋은거 같아요.
아울러 질문을 조금 드리고 싶은데요...

1. init repo 단계에서 보면 feature, release, hotfix 등의 branch 설정을 할수 있는 부분이 있는데요.
만약 해당 브랜치를 미리 생성하지 않고 실수로 OK를 누른 경우
1) 툴에서 자동으로 생성을 해주는건지(당장은 아니더라도 Finish Feature - Finish Release등의 절차를 밟아가는 동안이라도요)
2) 자동 생성은 아니고 설정한 브랜치 이름에 맞추어 사용자가 브랜치들을 생성해줘야 동작하는지
알고 싶습니다.

2. init repo 설정은 한번 ok 해버리면 플러그인설정에서도 다시 설정 못하던데 재설정 방법이 있는지 알고싶습니다.

3. 작업 다 마치고 finish feature 하는 순간에 fatal: Working tree contains unstaged changes. Aborting.
이라는 오류메세지가 뜨는데, 구글 검색해봐도 도통 원인을 모르겠습니다. 혹시 아시는 바가 있는지...
## 이건 자체 해결했습니다. 커밋 제대로 안하고 finish 를 눌러서 생긴오류였고 다 정리하고 finish feature 눌러주면 정상동작 하네요.

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

안녕하세요! 댓글 감사합니다

1-1) 브랜치는 init repo시에는 생성되지 않구요. 
start feature 단계에서 생성됩니다.
혹은 Task 생성시에 feature 브랜치가 자동 생성됩니다.

1-2) 브랜치는 사용자가 수동생성하진 않구요.
Gitflow 의 start feature를 통해 생성하시면 됩니다!

2. 제가 알기로는 init 레포 설정은 프로젝트에 종속적이라서 새로운 프로젝트에서 다시 해야할수있습니다
새 프로젝트를 생성하시면 되지 않을까 싶습니다!
(요건 확인이 필요하네요!)

**Comment by jojoldu** [comment]
작성자: 카엔

답변 감사드립니다! 프로젝트 종속적이라는 힌트 듣고는 방법을 찾았습니다. 이 플러그인 관해선 여기 글만큼 잘 설명해준 곳이 없어서 저같은 사람 또 나올까봐 달아둡니다.

프로젝트에 폴더에 보면 (탐색기나 파인더에서 '숨겨진 파일보기' 해주셔야 합니다) 
.git 라는 폴더에 config파일을 편집하면, 

[gitflow "branch"]
	master = main
	develop = ?????

이런 부분이 있습니다. 여기서 develop에 해당하는 부분에 넣고 싶은 브랜치 명을 적어주면 됩니다. 
같은 방식으로 다른 브랜치 이름이나 설정도 변경 가능합니다.

**Comment by Duckuism** [comment]
안녕하세요 동욱님!! 좋은 블로그글 감사합니다. 덕분에 trello 연동에 성공해서 유용하게 사용하고 있습니다.
그런데 task를 갖고 와서 branch를 gitflow로 생성할 때 실패하는 경우가 있는데요. 이런 경우에는 task가 비활성화되어서 같은 task를 다시 선택해도 gitflow 옵션이 나오지 않더라고요. 혹시 이런 경우에 undo하는 방법이 있을까요~!?

**Comment by Duckuism** [comment]
앗.. 그리고 clear current context라는 옵션은 task를 비활성화하겠다는 옵션으로 보면 되는 걸까요? 일단 task를 선택하고 branch를 만들면 shift+option+n 으로 눌렀을 때 항상 음영처리되어서 비활성화 되어있더라고요 ㅎㅎ;;
